az-core/src/main/java/azkaban/utils/ExecutorServiceUtils.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.utils;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Executor service related utilities.
+ */
+public class ExecutorServiceUtils {
```

Comment:
Since it's an util class, why not make gracefulShutdown static? What's the advantage of making this class injectable?

Reply:
I am glad you asked.

How would you test the code that uses this method if this method is made static?

Reply:
Some mocking lib allows mocking a static method.
https://stackoverflow.com/questions/21105403/mocking-static-methods-with-mockito

Reply:
I haven't used these libs. Do you think using these mocking libraries is better? If so, why?

Reply:
Discussed offline, static method needs a non-standard way of injecting and test. With dependency injection, we can declare an util class as singleton, which serves the same as an util class containing only static methods.